### PR TITLE
pull in forked changes to support multiple CI systems. Update readme.

### PR DESCRIPTION
**Pull request by maplebed**
This change completely rewrites the README, hopefully in a positive direction.

The changes to buildevents itself are mostly cleanup, with one added feature - looking for an environment variable to indicate which CI provider is in use and including that in events.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,59 +1,292 @@
...
+* installing the `buildevents` binary in your build environment
+* setting a number of environment variables for configuring the tool
+* choosing a unique trace identifier
+
+## Installation
+
+If you have a working go environment in your build, the easiest way to install `buildevents` is via `go install`.
+
+```
+go get https://github.com/honeycombio/buildevents/
```
Comment (toshok) [inline README.md]: go get => go install?

File: README.md [hunk 2]
```diff
@@ -1,59 +1,292 @@
...
-* `BUILDEVENT_APIHOST` overrides the API target for sending Honeycomb traces.  Default is `https://api.honeycomb.io/`
+There is one required environment variable; it will hold your Honeycomb API key (available at https://ui.honeycomb.io/account). If it is absent, events will not be sent to Honeycomb. Set `BUILDEVENT_APIKEY` to hold your API key.
+
+There are several other optional enviornment variables that will adjust the behavior of `buildevents`:
+
+* `BUILDEVENT_URL` the base URL back to the builds. This is used to construct a link back to the build log from within the trace. It will have the Trace ID appended to it. For example, in Travis-CI a URL like `https://travis-ci.org/honeycombio/buildevents/builds/` will combine with using the Build ID as the Trace ID and create a working link to the specific build from within the trace.
+* `BUILDEVENT_DATASET` sets the Honeycomb dataset to use. The default is `buildevents`
+* `BUILDEVENT_APIHOST` sets the API target for sending Honeycomb traces.  Default is `https://api.honeycomb.io/`
+* `BUILDEVENT_CIPROVIDER` if set, a field in all spans named `ci_provider` will contain this value
```
Comment (toshok) [inline README.md]: should we mention the auto-detection of travis-ci/circleci here?

File: README.md (hunk 1 above)
Comment (maplebed) [reply README.md]: Go get does install too! Can you do an install without a get? Will look.

File: main.go [hunk 3]
```diff
@@ -138,15 +144,22 @@ func handleCmd() error {
...
-	// TODO readme comments about setting Travis env vars for the apikey, and that it's the only required argument
 	apikey, _ := os.LookupEnv("BUILDEVENT_APIKEY")
 	dataset, _ := os.LookupEnv("BUILDEVENT_DATASET")
 	apihost, _ := os.LookupEnv("BUILDEVENT_APIHOST")
 	buildurl, _ := os.LookupEnv("BUILDEVENT_URL")
+	ciProvider, _ := os.LookupEnv("BUILDEVENT_CIPROVIDER")
+	if ciProvider == "" {
+		if _, present := os.LookupEnv("TRAVIS_BUILD_NUMBER"); present {
+			ciProvider = "Travis-CI"
+		} else if _, present := os.LookupEnv("CIRCLE_WORKFLOW_ID"); present {
```
[the remark this answers is not in the document]
Comment (maplebed) [reply main.go]: If it’s not, detection doesn’t work, but there’s no other downside, right? Perhaps a better response is to look at the list of available environment variables and choose one that will always be there. I’ll do that.

[the remark this answers is not in the document]
Comment (maplebed) [reply main.go]: Used `CIRCLECI` which will always be set to `true` when in a circleci build. While I was there a bunch of other environment variables looked useful so I added them too.

File: main.go [hunk 4]
```diff
@@ -162,9 +175,15 @@ func main() {
 	traceID := os.Args[2]
 
 	// add the build URL to all spans
-	libhoney.AddField("build_url", buildurl+traceID)
+	if buildurl != "" {
+		libhoney.AddField("build_url", buildurl+traceID)
```
[the remark this answers is not in the document]
Comment (maplebed) [reply main.go]: so turns out CircleCI and Travis-CI both provide a URL to the build as an environment variable. We don't need to hand build the URL.  Pulling this entirely.

**Comment by maplebed** [comment]
ok, made all these changes, but had an accident in my git pull / fetch / push flow and merged but then reset. This same content is getting pushed with #2 instead.

**Pull request merged**